pyuvdata/uvdata/uvdata.py

```diff
@@ -1398,34 +1436,35 @@ def _check_for_unprojected(self):
         if self.multi_phase_center:
             # Check and see if we have any unprojected objects, in which case
             # their w-values should be zeroed out.
-            nophase_dict = {
-                self.phase_center_catalog[name]["cat_id"]: self.phase_center_catalog[
-                    name
-                ]["cat_type"]
-                == "unprojected"
-                for name in self.phase_center_catalog.keys()
-            }
+            nophase_list = [
+                cat_id
+                for cat_id, cat_dict in self.phase_center_catalog.items()
+                if cat_dict["cat_type"] == "unprojected"
+            ]
 
-            # Use dict to construct a bool array
-            blt_mask = np.array(
-                [nophase_dict[idx] for idx in self.phase_center_id_array], dtype=bool
-            )
+            # Construct a bool mask
+            blt_mask = np.isin(self.phase_center_id_array, nophase_list)
         else:
             # If not multi phase center, we just need to check the phase type
             blt_mask = np.repeat(self.phase_type == "drift", self.Nblts)
 
         return blt_mask
 
-    def rename_phase_center(self, old_name, new_name):
+    def rename_phase_center(self, old_name=None, new_name=None, cat_id=None):
         """
         Rename a phase center/catalog entry within a multi phase center data set.
 
         Parameters
         ----------
-        old_name : str
-            Phase center name for the data to be renamed.
         new_name : str
             New name for the phase center.
+        old_name : str
+            Optional argument, phase center name for the target to be renamed. Note that
+            either `old_name` or `cat_id` must be specified.
+        cat_id : int or list of int
+            Optional argument, phase center IDs number for the target to be renamed.
+            Note that either `old_name` or `cat_id` must be specified.
```

Comment:
This order should match the function signature order (swap old_name and new_name in either the signature or docstring).